state-synchronizer/src/coordinator.rs

```diff
@@ -866,16 +867,14 @@ impl<T: ExecutorProxyTrait> SyncCoordinator<T> {
                 None => TargetType::HighestAvailable {
                     timeout_ms: self.config.long_poll_timeout_ms,
                 },
-                Some(sync_req) => {
-                    if sync_req.target.ledger_info().version() <= known_version {
-                        debug!(
-                            "[state sync] Reached version {}, no need to send more requests",
-                            known_version
-                        );
-                        return Ok(());
-                    }
-                    TargetType::TargetLedgerInfo(sync_req.target.clone())
+                Some(sync_req) if sync_req.target.ledger_info().version() <= known_version => {
+                    debug!(
+                        "[state sync] Reached version {}, no need to send more requests",
+                        known_version
+                    );
+                    return Ok(());
                 }
+                Some(sync_req) => TargetType::TargetLedgerInfo(sync_req.target.clone()),
```

Comment:
I'm not sure if this is more or less clear than the original.. (but that's my personal preference -- I'm not a big fan of overlapping match guards if they can be otherwise avoided)

Reply:
personally I find this less clear and more error-prone.. what is wrong with the current way it is written?

Reply:
when using matching you have access to guards for exactly this type of scenarios, which is supposed to make the code clearer. We don't really have guidelines about this so I don't care much, but it's clearer to me.

Maybe some rust experts should think about the place of this pattern in our codebase :) @bmwill @sunshowers

Reply:
Don't have a strong opinion either way, but this is fairly similar to how match is used in FP languages so it's not immediately objectionable to me.

Reply:
thanks for chiming in! Indeed, if you're unfamiliar with FP this might look weird, at the same time Rust looks weird to anyone unfamiliar with FP :D